src/universal/modules/meeting/components/CopyShortLink/CopyShortLink.js

```diff
@@ -1,82 +1,61 @@
-import appTheme from 'universal/styles/theme/appTheme';
-import ui from 'universal/styles/ui';
-import {textOverflow} from 'universal/styles/helpers';
 import voidClick from 'universal/utils/voidClick';
 
-const inlineBlock = {
-  display: 'inline-block',
-  height: '1.75rem',
-  lineHeight: '1.75rem',
-  verticalAlign: 'top'
-};
+class CopyShortLink extends Component {
+  static propTypes = {
+    url: PropTypes.string
+  };
 
-const CopyShortLink = (props) => {
-  const {styles, url} = props;
-  return (
-    <CopyToClipboard text={url}>
-      {/* TODO: prevent navigation and show a “Copied!” message inline or toast */}
-      <a
-        className={css(styles.link)}
-        href={url}
-        onClick={voidClick}
-        title={`Copy link to meeting: ${url}`}
-      >
-        <span className={css(styles.linkText)}>{url}</span>
-        <span className={css(styles.icon)}>
-          <FontAwesome
-            name="copy"
-            style={inlineBlock}
-          />
-        </span>
-      </a>
-    </CopyToClipboard>
-  );
-};
+  constructor(props) {
+    super(props);
+    this.confirmationTimeout = null;
+  }
 
-CopyShortLink.propTypes = {
-  styles: PropTypes.object,
-  url: PropTypes.string
-};
+  state = {
+    confirmingCopied: false
+  };
 
-const styleThunk = () => ({
-  link: {
-    backgroundColor: appTheme.palette.cool10l,
-    borderRadius: '.375rem',
-    display: 'block',
-    fontSize: 0,
-    lineHeight: '1.75rem',
-    margin: '0 auto',
-    maxWidth: '24rem',
-    padding: '.625rem .5rem',
-    textAlign: 'center',
-    textDecoration: 'none !important',
+  componentWillUnmount() {
+    this.clearConfirmationTimeout();
+  }
 
-    ':hover': {
-      backgroundColor: appTheme.palette.cool20l
-    },
-    ':focus': {
-      backgroundColor: appTheme.palette.cool20l
+  clearConfirmationTimeout = () => {
+    if (this.confirmationTimeout) {
+      clearTimeout(this.confirmationTimeout);
     }
-  },
-
-  linkText: {
-    ...inlineBlock,
-    ...textOverflow,
-    fontSize: appTheme.typography.s6,
-    maxWidth: '20rem'
-  },
-
-  icon: {
-    ...inlineBlock,
-    fontSize: ui.iconSize2x,
-    marginLeft: '.5rem'
+  };
+
+  confirmCopied = () => {
+    this.clearConfirmationTimeout();
+    this.confirmationTimeout = setTimeout(() => {
+      this.setState({ confirmingCopied: false });
+    }, 1500);
+    this.setState({ confirmingCopied: true });
+  };
+
+  render() {
+    const {url} = this.props;
+    const {confirmingCopied} = this.state;
+    const callToAction = 'Copy link to meeting';
+    return (
+      <CopyToClipboard text={url} onCopy={this.confirmCopied}>
+        <Button
+          aria-label={callToAction}
+          buttonStyle="inverted"
+          colorPalette="cool"
+          disabled={confirmingCopied}
+          title={callToAction}
+          icon={confirmingCopied ? 'check' : 'copy'}
+          iconPlacement="left"
+          label={confirmingCopied ? 'Copied!' : url}
+          onClick={voidClick}
```

Comment:
I think `size="small"` may be the winner for this button. I think I like the button in this context. I like how you are handling the icon and label as feedback. Feel free to upgrade to a tooltip for feedback if you want. I think the tooltip might be nice (perhaps set to close after a brief amount of time) while the button maintains the original URL text.

Reply:
I agree that the tooltip would be nice. I'm going to cherry-pick this commit onto a sprint branch off of the `epic7` branch, and follow up there.

Reply:
👍 👍